taesch/lib/view/screen/settings_screen.dart

```diff
@@ -23,20 +23,121 @@ class _SettingsScreenState extends State<SettingsScreen> {
+                logger.log(
+                    level: LogLevel.info,
+                    logMessage: LogMessage(
+                        message:
+                        "switched ${widget._vm.switchTitle} button to $value"));
+                setState(() {
+                  ThemeController.of(context).darkTheme = value;
+                });
+              },
+              secondary: const Icon(Icons.sunny),
+          ),
+
+          SwitchListTile(
+            title: Text(widget._vm.permissionSwitchTitle),
+            value: widget._vm.repository.geolocationTools.geoLocationPermissionGranted(),
+            onChanged: (bool permitted) async{
+              logger.log(
+                  level: LogLevel.info,
+                  logMessage: LogMessage(
+                      message:
+                      "switched ${widget._vm.permissionSwitchTitle} button to $permitted"));
+
+              var gt = widget._vm.repository.geolocationTools;
+
+              if (!permitted){
+                /*
+                IMPORTANT: this is not an actual denial of the service, as the OS would understand it.
+                It is only a denial interpreted by the App, which ultimately achieves the same result of
+                not using the GPS Location.
+                 */
+                gt.denyGeoLocationPermission();
+
+              }else{
+                // at this point I know the permission must be denied
+                bool permDenial = gt.geoLocationPermissionIsPermanentlyDenied();
+                if (!permDenial){
+                  await gt.handleLocationPermission();
+                  if (!gt.geoLocationPermissionGranted()){
+                    // don't allow toggle
+                  }
+                }else{
+                  // pop-up
+                  showDialog(
+                      context: context,
+                      builder: (ctx) => AlertDialog(
+                          title: const Text("Location Permanently Denied"),
+                          content: const Text("You have permanently denied the usage of the GPS services.\n"
+                              "In order to allow the app to use them, you have to manually go to your phone's \n"
+                              "Settings -> Apps -> 'täsch' -> Permissions and allow the permission for "
+                              "the usage of your location. Thanks ;)"),
+
+                          actions: [
+                            TextButton(
+                                onPressed: (){
+                                  Navigator.pop(ctx);
+                                },
+                                child: const Text("OK")
+                            )
+                          ]
+                      )
+                  );
+                  // don't allow toggle
+                }
+
+              }
+              setState((){}); // because the switch is set based on the .geoLocationPermGranted() property,
```

Comment:
Ein leeres setState() hat keine Funktionalität.

2 Möglichkeiten:

1. Löschen dieses setState()
2. Funktionialität in das setState() hinzufügen

Reply:
ja das ist so gewollt. Das setState() soll den Screen nochmal neu laden

Reply:
Funktionalität hinzugefügt